incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -151,13 +166,32 @@ func (v *Hierarchy) checkForest(hc *api.HierarchyConfiguration) ([]authzReq, adm
 	// have visibility into its ancestry and descendents, and this check can only fail if the new
 	// parent conflicts with something in the _existing_ hierarchy.
 	if reason := ns.CanSetParent(newParent); reason != "" {
-		return nil, deny(metav1.StatusReasonConflict, "Illegal parent: "+reason)
+		return deny(metav1.StatusReasonConflict, "Illegal parent: "+reason)
 	}
 
-	// TODO: requiredChild check: https://github.com/kubernetes-sigs/multi-tenancy/issues/99
+	// Prevent changing parent of a required child
+	if ns.RequiredChildOf != "" && ns.RequiredChildOf != newParent.Name() {
+		reason := fmt.Sprintf("Cannot set the parent of %q to %q because it's a required child of %q", ns.Name(), newParent.Name(), ns.RequiredChildOf)
+		return deny(metav1.StatusReasonConflict, "Illegal parent: "+reason)
+	}
 
-	// The structure looks good. Get the list of namespaces we need authz checks on.
-	return v.needAuthzOn(curParent, newParent), allow("")
+	return allow("")
+}
+
+func (v *Hierarchy) checkRequiredChildren(ns *forest.Namespace, requiredChildren []string) admission.Response {
+	for _, child := range requiredChildren {
+		cns := v.Forest.Get(child)
+		// Early exit if we find the newly-added requiredChild
+		if !cns.Exists() {
+			return allow("RequiredChild namespace is just created")
+		}
+		if cns.Parent() != ns && (cns.Parent() != nil || cns.RequiredChildOf != ns.Name()) {
```

Comment:
This might be easier to read in the reverse:

```
// If this is already a child, or is about to be, no problem
if cns.Parent() == ns || (cns.Parent() == nil && cns.RequiredChildOf == ns.Name()) {
  continue
}
```